290.py

```diff
@@ -0,0 +1,19 @@
+class Solution:
+    def wordPattern(self, pattern: str, s: str) -> bool:
+        strs = s.split()
+
+        if len(pattern) != len(strs):
+            return False
+
+        str_to_pattern = dict()
+        pattern_to_str = dict()
+
+        for idx, (s, p) in enumerate(zip(strs, pattern)):
+            if (s in str_to_pattern and str_to_pattern[s] != pattern[idx]) or \
```

Comment:
I would break this into two `if` statements:

```python
def wordPattern(self, pattern: str, s: str) -> bool:
    p2w = {}
    w2p = {}
    words = s.split()
    
    if len(pattern) != len(words):
        return False
    
    for p, word in zip(pattern, words):
        if p in p2w and p2w[p] != word:
            return False
        
        if word in w2p and w2p[word] != p:
            return False
        
        if p not in p2w:
            p2w[p] = word
            
        if word not in w2p:
            w2p[word] = p
            
    return True